### PR TITLE
doc: Updated -blocksonly documentation

### PR DESCRIPTION
**Pull request by birtony**
Updated documentation for `blocksonly` according to #17294.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -366,7 +366,7 @@ void SetupServerArgs()
     gArgs.AddArg("-blocknotify=<cmd>", "Execute command when the best block changes (%s in cmd is replaced by block hash)", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
 #endif
     gArgs.AddArg("-blockreconstructionextratxn=<n>", strprintf("Extra transactions to keep in memory for compact block reconstructions (default: %u)", DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-blocksonly", strprintf("Whether to reject transactions from network peers. Transactions from the wallet, RPC and relay whitelisted inbound peers are not affected. (default: %u)", DEFAULT_BLOCKSONLY), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-blocksonly", strprintf("Whether to reject transactions from network peers. Transactions from the wallet, RPC and relay whitelisted inbound peers are affected. (default: %u)", DEFAULT_BLOCKSONLY), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (MarcoFalke) [inline src/init.cpp]: This is not true. It depends on the parameter interaction. I'd rather describe that here.

Also you'd need to update doc/low_bandwith.md or whatever it is called

Comment (birtony) [reply src/init.cpp]: Hey @MarcoFalke, I am sorry for my mistake, and thanks for your comments. Should I put it as the following?
> Whether to reject transactions from network peers. Transactions from the wallet, RPC and relay whitelisted inbound peers might be affected depending on the parameter interaction.

Comment (birtony) [reply src/init.cpp]: @MarcoFalke I have made an update which includes your remarks in a333b82. Please, let me know if it is correct?

**Comment by fanquake** [comment]
@birtony You need to remove any trailing whitespace and [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits).

File: doc/reduce-traffic.md [hunk 2]
```diff
@@ -48,3 +48,5 @@ Be reminded of the effects of this setting.
   wallet is loaded or if you use the node to broadcast transactions.
 - It makes block propagation slower because compact block relay can only be
   used when transaction relay is enabled.
+- Transactions from the wallet, RPC and relay whitelisted inbound peers 
+  might be affected depending on the parameter interaction.
```
Comment (MarcoFalke) [inline doc/reduce-traffic.md]: As a user reading this, there is no way to extract any useful information. This should exactly explain which parameters can interact.

**Comment by fanquake** [comment]
No longer up for grabs (#18391).

**Pull request closed without merging**